### PR TITLE
Cross-compile to netcore50

### LINKED ISSUE #4802: Latest RC2 packages do not install in latest UWP builds
The errors are currently in our dependencies, so there is likely nothing to do on our side other than test and report issues to the appropriate teams. I've reported the current issue with the `System.AppContext` package to .NET Core folks.

> System.AppContext 4.1.0-rc2-23826 provides a compile-time reference assembly for System.AppContext on UAP,Version=v10.0, but there is no run-time assembly compatible with win10-arm.
At line:1 char:1
install-package Microsoft.EntityFrameworkCore.Sqlite -pre
~~~~~~~~~~~~~
CategoryInfo : NotSpecified: (:) [Install-Package], Exception
FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix powershell commands.

Fix #4802 

Related pull requests for dependencies: 

https://github.com/aspnet/Common/pull/107
https://github.com/aspnet/Caching/pull/178
https://github.com/aspnet/Options/pull/137
https://github.com/aspnet/PlatformAbstractions/pull/32
https://github.com/aspnet/DependencyInjection/pull/399
https://github.com/aspnet/Logging/pull/403
https://github.com/aspnet/Microsoft.Data.Sqlite/pull/242

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/StartupInvoker.cs [hunk 1]
```diff
@@ -100,12 +102,14 @@ private object Invoke(Type type, string[] methodNames, IServiceCollection servic
 
         protected virtual IServiceCollection ConfigureHostServices([NotNull] IServiceCollection services)
         {
+#if !NETCORE50
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/StartupInvoker.cs]: For possibly improved logistics, consider merging the changes from https://github.com/aspnet/EntityFramework/pull/5037 as well, since UWP does not work until the commands work.

**Comment by divega** [comment]
I am not able to validate if the changes to project.json files seem correct, but the rest LGTM.

**Comment by natemcmaster** [comment]
cc @pranavkm @davidfowl can someone review/signoff on the project.json changes?

File: src/Microsoft.EntityFrameworkCore.Commands/tools/OperationHandlers.cs [hunk 2]
```diff
@@ -32,17 +38,17 @@ public partial class OperationResultHandler : IOperationResultHandler
         private string _errorMessage;
         private string _errorStackTrace;
 
-        public virtual int Version => 0;
+        public virtual int Version { get { return 0; } }
```
Comment (pranavkm) [inline src/Microsoft.EntityFrameworkCore.Commands/tools/OperationHandlers.cs]: Why change these?

Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Commands/tools/OperationHandlers.cs]: This file also needs to be compiled in c# 5.0 by powershell. See #5036

Comment (pranavkm) [inline src/Microsoft.EntityFrameworkCore.Commands/tools/OperationHandlers.cs]: Shouldn't that go in a separate PR? If the hope here is that eventually we won't have to cross compile for netcore, we could revert this change in entirety.

File: src/Microsoft.EntityFrameworkCore.Relational/project.json [hunk 3]
```diff
@@ -38,8 +40,29 @@
...
+    },
+    "netcore50": {
+      "dependencies": {
+        "Microsoft.NETCore.Platforms": {
+          "type": "build",
+          "version": "1.0.1-*"
+        },
+        "Microsoft.CSharp": "4.0.0",
+        "System.Data.Common": "4.0.0",
+        "System.Text.RegularExpressions": "4.0.10",
```
Comment (pranavkm) [inline src/Microsoft.EntityFrameworkCore.Relational/project.json]: Why are some of these `type: build` and not others?

**Comment by pranavkm** [comment]
Looks :shipit: to me. Might want to get @davidfowl \ @anurse \ @pakrym to look at this in case they spot issues working with the CLI.

File: src/Microsoft.EntityFrameworkCore.Relational/project.json (hunk 3 above)
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Relational/project.json]: In short, to make dotnet-restore pass but they don't need to be in the nuspec because these packages will only ever be installed alongside Microsoft.NETCore.UniversalWindowsPlatform.

"build" dependencies are there only to upgrade the version of the package dotnet-restore pulls. Without this (for example without "System.Dynamic.Runtime" build dependency), nuget fails. "Could not find runtime-assembly for win7-x64 for System.Dynamic.Runtime 4.0.0.

The same is true of including ms.netcore.platforms. It doesn't need to be in the nuspec.
